src/controllers/audience/audienceController.ts

```diff
@@ -149,30 +151,42 @@ export class AudienceController {
 
 	getAudienceType() {
 		return async (req: Request, res: Response) => {
-			res.render('page/course/audience/audience-type')
+			res.render('page/course/audience/audience-type', {AudienceType: Audience.Type})
 		}
 	}
 
 	setAudienceType() {
 		return async (req: Request, res: Response) => {
 			const data = {...req.body}
 			const errors = await this.audienceValidator.check(data, ['audience.type'])
-			const audience = this.audienceFactory.create(data)
+			const audienceFromData = this.audienceFactory.create(data)
 
 			if (errors.size > 0) {
-				req.session!.sessionFlash = {errors, audienceName: audience.name}
+				req.session!.sessionFlash = {errors, audienceName: audienceFromData.name}
 				req.session!.save(() => {
 					res.redirect(`/content-management/courses/${req.params.courseId}/audiences/type`)
 				})
 			} else {
-				const savedAudience = await this.learningCatalogue.createAudience(req.params.courseId, audience)
-				req.session!.sessionFlash = {audience: savedAudience}
-				req.session!.save(() => {
-					res.redirect(
-						// prettier-ignore
-						`/content-management/courses/${req.params.courseId}/audiences/${savedAudience.id}/configure`
+				let savedAudienceId: string
+				if (res.locals.audience) {
+					this.audienceService.updateAudience(
```

Comment:
MAINTAIN + READABILITY - Is there no concept of promises here rather callbacks?
`updateAudience` takes in a callback and then audience fed in again to call the same method. Hard to follow and see the execution in code let alone debugging.

The ideal scenario for readability.

```
audienceService.doSomething()
.then().doSomething()
.then().doOther()
.success(print)
.error(message)```

Reply:
made changes removing the lambda - not using Futures, since it's not needed anymore due to the changes